src/ert/_c_wrappers/job_queue/run_status.py

```diff
@@ -0,0 +1,36 @@
+from dataclasses import dataclass
+from typing import Optional
+
+
+@dataclass
+class RunStatus:
+    RUNNING = "RUNNING"
+    FINISHED = "FINISHED"
```

Comment:
This class is a bit difficult to understand as it combines the concept of a Enum (RUNNING, FINISHED, CANCELED, FAILED) but is actually a dataclass. I think it would make sense for to actually hold the script that is cancelled as the "cancel" method doesn't actually cancel the job it just sets the status.

Reply:
Maybe have a 
```
class RunStatus(Enum):
   RUNNING = auto()
   FINISHED = auto()
   CANCELED = auto()
   FAILED = auto()
```

and a separate:
```
@dataclass
class RunningJob:
    status: RunStatus = RunStatus.RUNNING
    stdoutdata: str = ""
    stderrdata: str = ""
    script: ErtScript

    def start(self, args):
        self.status = RunStatus.RUNNING
        self.script.run(args)

    def cancel(self):
        self.status = RunStatus.CANCELED
        self.script.cancel()
    ...
```

Reply:
I understand that RunStatus might be a bit hard to understand initially, maybe having it as a @dataclass is also adding to the confusion, it is designed to be a minimal internal status tracker for other objects, like ErtScript and Workflow. The aim is not to add another wrapper,  I think there are enough wrapper classes already with WorflowJob and Plugin and workflowRunner and maybe some other class that I don't remember now.

Reply:
It might be true that Workflow is doing too many things it could just be a list of jobs, but that is a bigger redesign than I planned on tackling with these changes so we can just do this in the future when it becomes necessary.

Reply:
Ok :)